plugins/source/azure/client/services/subscriptions.go

```diff
@@ -10,35 +10,40 @@ import (
 	"github.com/Azure/go-autorest/autorest"
 )
 
-type Subscriptions struct {
+type SubscriptionsClient struct {
 	SubscriptionID string
-	Subscriptions  SubscriptionsClient
-	Tenants        TenantsClient
+	Subscriptions  SubscriptionsSubscriptionsClient
+	Locations      SubscriptionsLocationsClient
+	Tenants        SubscriptionsTenantsClient
 }
 
-type SubscriptionsClient interface {
+type SubscriptionsSubscriptionsClient interface {
 	NewListPager(options *armsubscriptions.ClientListOptions) *runtime.Pager[armsubscriptions.ClientListResponse]
+}
+
+type SubscriptionsLocationsClient interface {
 	NewListLocationsPager(subscriptionId string, options *armsubscriptions.ClientListLocationsOptions) *runtime.Pager[armsubscriptions.ClientListLocationsResponse]
 }
 
-type TenantsClient interface {
+type SubscriptionsTenantsClient interface {
 	NewListPager(options *armsubscriptions.TenantsClientListOptions) *runtime.Pager[armsubscriptions.TenantsClientListResponse]
 }
 
-func NewSubscriptionsClient(subscriptionId string, auth autorest.Authorizer, azCred azcore.TokenCredential) (Subscriptions, error) {
+func NewSubscriptionsClient(subscriptionId string, auth autorest.Authorizer, azCred azcore.TokenCredential) (SubscriptionsClient, error) {
 	s, err := armsubscriptions.NewClient(azCred, nil)
 	if err != nil {
-		return Subscriptions{}, fmt.Errorf("failed to create subscriptions client: %w", err)
+		return SubscriptionsClient{}, fmt.Errorf("failed to create subscriptions client: %w", err)
 	}
 
 	t, err := armsubscriptions.NewTenantsClient(azCred, nil)
 	if err != nil {
-		return Subscriptions{}, fmt.Errorf("failed to create tenants client: %w", err)
+		return SubscriptionsClient{}, fmt.Errorf("failed to create tenants client: %w", err)
 	}
 
-	return Subscriptions{
+	return SubscriptionsClient{
 		SubscriptionID: subscriptionId,
 		Subscriptions:  s,
 		Tenants:        t,
+		Locations:      s,
```

Comment:
`account/locations.go` (see [here](https://github.com/cloudquery/cloudquery/blob/973af8b82b59e0c3af7ae031c866888fea1edbca/plugins/source/azure/resources/services/account/locations.go#L134)) used the `SubscriptionsClient` which was a bit confusing.

I moved it under `subscriptions/locations.go`  and added a separate `SubscriptionsLocationsClient` to make code generation easier and keep everything in a similar structure